src/arrayMethodJoin.js

```diff
@@ -5,7 +5,67 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let string = '';
+
+    if (this.length === 0) {
+      return string;
+    }
+
+    if (this.length === 1) {
+      string += this[0];
+
+      return string;
+    }
+
+    for (let i = 0; i <= this.length - 1; i++) {
+      if ((this[i] === undefined || this[i] === null)
```

Comment:
too many checks now
you only need to check for 2 main things
1) if this[i] !== null && this[i] !== undefined, then you can concatenate the next element
2) if this is not the last element of the string then you add the separator

that's all, try to refactor your code with these minimum checks in your mind